Code/Tools/SceneAPI/SceneCore/Events/ExportProductList.cpp

```diff
@@ -106,6 +106,14 @@ namespace AZ
                 AZ_Assert(!lod.has_value() || lod < 16, "Lod value has to be between 0 and 15 or disabled.");
 
                 size_t index = m_products.size();
+                auto FindProduct = [productFilename = AZ::IO::PathView(filename)](const ExportProduct& exportProduct)
+                {
+                    return AZ::IO::PathView(exportProduct.m_filename) == productFilename;
+                };
+                if (auto productIt = AZStd::find_if(m_products.begin(), m_products.end(), FindProduct); productIt != m_products.end())
+                {
+                    return *productIt;
+                }
```

Comment:
So, for files with 2000+ submeshes, the O(n^2) nature of this is really going to explode. :( I'm not loving this as a solution. Maybe turn m_products into an unordered_set and not even bother to do the find, just do an emplace? idk if the order is important here or not.

Reply:
Yeah, i'm not sure if order matters either. Another option maybe is to keep an additional unordered set just to track duplicate filenames instead and leave `m_products` as is?

Reply:
The order shouldn't matter here.
The products all need to be unique based on the Asset SubID anyway.

Reply:
Maybe a different approach could be alter where the [duplicate check ](https://github.com/o3de/o3de/blob/cb0fbb84410ec44086c9358233a2e3f74fbad9e2/Code/Tools/AssetProcessor/native/resourcecompiler/rcjob.cpp#L618-L628)is occuring and strip out the duplicates there and instead report a warning?

Reply:
If you do that, you'd need to make sure you were only catching the case of duplicate product names, and not just duplicate sub IDs. Otherwise it will cause products that currently fail from subID collisions into successes-with-warnings which isn't what we would want.

I'm mixed on the idea of reporting a warning. I conceptually like the idea of keeping a warning around for the case where it's trying to emit duplicate products, since that's still pointing to a root cause problem that's going unresolved. However, it will be a completely unactionable warning that emits for every exported FBX under Snap, so it doesn't feel valuable to the end user. I guess I'd vote for no warning, but a GHI that explains the problem and points out the root cause is unsolved.

Reply:
Then lets go with using an unordered_set instead then..

Reply:
If we go with an unordered_set, you only need to add a hash function that hashes the product filename or an `operator size_t` [member function](https://github.com/o3de/o3de/blob/cb0fbb84410ec44086c9358233a2e3f74fbad9e2/Code/Framework/AzCore/AzCore/std/hash.h#L37-L41) which would hash the product filename.
The [ExportProduct](https://github.com/o3de/o3de/blob/cb0fbb84410ec44086c9358233a2e3f74fbad9e2/Code/Tools/SceneAPI/SceneCore/Events/ExportProductList.h#L24) class would also need an `operator==`

Reply:
Yeah, I just realized that, maybe all I need is to track the product paths separately in an unordered set and keep the product vector as is.